packages/api/src/modules/providers/scdl/scdl.service.ts

```diff
@@ -65,6 +65,79 @@ export class ScdlService {
     parseCsv(fileContent: Buffer, delimiter = ";", quote: string | boolean = '"') {
         return ScdlGrantParser.parseCsv(fileContent, delimiter, quote);
     }
+
+    /**
+     *
+     * @param slug Producer slug
+     * @param entities Entities parsed from the current import
+     * @returns Last exercise number from the import
+     * @throws Error if the import has less exercises than the existing ones in the database
+     */
+    async validateAndGetLastExerciseGrants(slug: string, entities: ScdlStorableGrant[]) {
+        const exercises: Set<number> = entities.reduce((acc, entity) => {
+            return acc.add(entity.exercice);
+        }, new Set<number>());
+
+        if (exercises.size === 0) {
+            throw new Error("You must provide an exercise to clean producer's data before import");
+        }
+
+        const exercisesArray = [...exercises]; // transform Set to Array
+        const documents = await miscScdlGrantPort.findBySlugOnPeriod(slug, exercisesArray);
+
+        console.log(`There are currently ${documents.length} documents for producer ${slug}`);
+        console.log(`The new import contains ${entities.length} entities`);
+
+        exercisesArray.forEach(exercise => {
+            console.log(documents.length, entities.length, exercise);
+            if (
+                documents.filter(doc => doc.exercice === exercise).length >
+                entities.filter(entity => entity.exercice === exercise).length
+            ) {
+                throw new Error(
+                    `You are trying to import less grants for exercise ${exercise} than what already exists in the database for producer ${slug}.`,
+                );
+            }
+        });
+
+        const lastExercise = exercisesArray.sort((a, b) => a - b).at(-1) as number;
```

Comment:
```suggestion
        const lastExercise = exercisesArray.sort((a, b) => b - a)[0] as number;
```
Je crois que c'est pareil mais plus lisible

Reply:
C'est trop subjectif. Je préfère la première solution, qui pour moi est plus intuitive

Reply:
Si tu préfères ok, je suis d'accord que c'est subjectif